odoo/tools/translate.py

```diff
@@ -997,76 +1003,110 @@ def babel_extract_terms(fname, path, root, extract_method="python", trans_type='
-        trans = Translation._get_source(name, type, lang, source) if lang else ""
-        out.append((module, type, name, id, source, encode(trans) or '', comments))
-    return out
+    def export_translatable_resources(self):
+        """ Export translations for static resources """
+
+        # Also scan these non-addon paths
+        for bin_path in ['osv', 'report', 'modules', 'service', 'tools']:
+            self._path_list.append((os.path.join(config['root_path'], bin_path), True))
+        # non-recursive scan for individual files in root directory but without
+        # scanning subdirectories that may contain addons
+        self._path_list.append((config['root_path'], False))
+        _logger.debug("Scanning modules at paths: %s", self._path_list)
+
+        for (path, recursive) in self._path_list:
+            _logger.debug("Scanning files of modules at %s", path)
+            for root, dummy, files in walksymlinks(path):
+                for fname in fnmatch.filter(files, '*.py'):
+                    self._babel_extract_terms(fname, path, root,
+                                              extract_keywords={'_': None, '_lt': None})
+                # Javascript source files in the static/src/js directory, rest is ignored (libs)
+                if fnmatch.fnmatch(root, '*/static/src/js*'):
+                    for fname in fnmatch.filter(files, '*.js'):
+                        self._babel_extract_terms(fname, path, root, 'javascript',
+                                                  extra_comments=[WEB_TRANSLATION_COMMENT],
+                                                  extract_keywords={'_t': None, '_lt': None})
+                # QWeb template files
+                if fnmatch.fnmatch(root, '*/static/src/xml*'):
+                    for fname in fnmatch.filter(files, '*.xml'):
+                        self._babel_extract_terms(fname, path, root, 'odoo.tools.translate:babel_extract_qweb',
+                                                  extra_comments=[WEB_TRANSLATION_COMMENT])
+                if not recursive:
+                    # due to topdown, first iteration is in first level
+                    break
+
+    def read_rows(self, lang=None):
+        """ Export ir.translation records for all exported records
+
+        :param lang: language code to retrieve the translations
+                     retrieve source terms only if not set
+        """
+        out = []
+        # translate strings marked as to be translated
+        Translation = self.env['ir.translation']
+        for module, source, name, res_id, ttype, comments in sorted(self._to_translate):
+            trans = Translation._get_source(name, ttype, lang, source) if lang else ""
+            out.append((module, ttype, name, res_id, source, encode(trans) or '', comments))
+        return out
 
 
 def trans_load(cr, filename, lang, verbose=True, module_name=None, context=None):
```

Comment:
What about making parameters `create_empty_translation` and `overwrite` explicit here, too?